langchain4j-core/src/main/java/dev/langchain4j/model/chat/request/ChatRequest.java

```diff
@@ -16,12 +16,14 @@
 public class ChatRequest {
 
     private final List<ChatMessage> messages;
-    private final List<ToolSpecification> toolSpecifications; // TODO add section with tools?
+    private final List<ToolSpecification> toolSpecifications; // TODO introduce "tools" section?
+    private final ToolChoice toolChoice;
     private final ResponseFormat responseFormat;
 
     private ChatRequest(Builder builder) {
         this.messages = new ArrayList<>(ensureNotEmpty(builder.messages, "messages"));
         this.toolSpecifications = copyIfNotNull(builder.toolSpecifications);
+        this.toolChoice = builder.toolChoice; // TODO set AUTO by default? only if toolSpecifications are present? validate: can be set only when tools are defined
```

Comment:
Yeah in the builder, we should probably throw an error if toolChoice=ANY and there are no tool specs

Reply:
I was considering leaving `toolChoice` `null` by default. This way we will just pass-through what user has configured to the LLM and it will take care of validating. Not sure it is the best strategy though. WDYT?

Reply:
Allowing a null in an enum field generally sounds like an anti-pattern to me (if it leads to a different behavior, it should have a separate value in the enum). 

But I'm not sure if we need to differentiate between AUTO (where the model should receive `toolChoice = auto` explicitly) and something where we don't specify a `toolChoice` param to the model at all. But if we do need to differentiate  it, it should be a new value of the enum (`UNSPECIFIED`?) but not null.

Reply:
I see, good point. `null` usually means `AUTO` (since LLM APIs have a default value of `AUTO`). We can set it to `AUTO` by default, but then when creating the specific LLM API request (in each integration) we will have to remove it if there are no tools specified, otherwise LLM API will fail... That is why I was considering leaving it null by default.

Reply:
Another option I was considering is to extract `tools` and `toolChoice` into a separate object (e.g. `ToolConfig` or something). So if tools are set, tool choice can be set as well. If tools are not set, there is no need in toolChoice. But this will make `ChatRequest` builder more nested/complicated...

Reply:
Then I think the easiest way would be to add a value named `UNSPECIFIED`, make that the default, and it would mean that the model integration code should not send any `tool_choice` parameter to the model.

Reply:
Then this will users will be forced to explicitly set `AUTO` when tools are present. Not only more inconvenient, but also differs in behaviour from all LLM APIs...

Reply:
Makes sense, I will try to wrap them into the `ToolConfig` and see if it works